src/output-json.c

```diff
@@ -77,6 +77,86 @@
 
 #define MAX_JSON_SIZE 2048
 
+#ifdef HAVE_GEOIP
+#include <maxminddb.h>
+
+static MMDB_s mmdb;
+static int mmdb_status = MMDB_FILE_OPEN_ERROR;
+
+void GeoIPGet(JsonBuilder *js, const MMDB_s *const mmdb, const char *ip_address, const char *key);
+
+inline void GeoIPSetString(JsonBuilder *js, MMDB_entry_data_s *entry_data, const char *key)
+{
+    char *str = NULL;
+    if (entry_data->has_data && entry_data->utf8_string != NULL) {
+        str = SCStrndup((char *)entry_data->utf8_string, entry_data->data_size);
```

Comment:
This isn't needed. The output is staged into a buffer into which the string will be copied.